WirescaleCli/Program.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Microsoft.Extensions.Configuration;
+
+namespace WirescaleCli;
+
+public class Program
+{
+    public static async Task Main(string[] args)
+    {
+        var configurationBuilder = new ConfigurationBuilder();
+        configurationBuilder.SetBasePath(Directory.GetCurrentDirectory())
+            .AddJsonFile("appsettings.json", optional: false);
+
+        var configuration = configurationBuilder.Build();
+
+        var accessTokenProvider = new AcessTokenProvider(configuration["Auth0:Domain"], configuration["Auth0:ClientId"], configuration["Auth0:RedirectUri"]);
+        var accessToken = await accessTokenProvider.GetAccessToken();
+
+        var wireguardKeyPairGenerator = new WireguardKeyPairGenerator();
+        var wireguardKeyPair = wireguardKeyPairGenerator.Generate();
+
+        var wirescaleApiClient = new WirescaleApiClient(configuration["WirescaleApiBaseUri"]);
+        try
+        {
+            var wireguardPeerRegistration = await wirescaleApiClient.RegisterPublicKey(accessToken, wireguardKeyPair);
+
+            var wireguardConfigurationWriter = new WireguardConfigurationWriter("wg-wirescale.conf");
+            wireguardConfigurationWriter.Write(wireguardKeyPair, wireguardPeerRegistration);
+
+            Console.WriteLine("You are now logged in.");
+        }
+        catch (HttpRequestException exception)
+        {
+            Console.WriteLine($"Failed to call the Wirescale API. Status code: {exception.StatusCode}");
```

Comment:
This is a very generic error message and some status code which doesn't feel very user-friendly tbh. How would we improve it so users don't get confused when things don't work and know what to do?

Reply:
I've added more specific error messages for the various known types of errors that can occur here. The API has also been updated to better support the more specific error messages.

In this case, there's little the user can do to resolve the issue (as they provide no input and authorization/access token issues are handled earlier). However, I did add an error message for when the Wirescale API is not running (e.g. a `SocketException`) which will likely be helpful for at least people using this CLI during the challenge :)

https://github.com/runesoerensen/wirescale/pull/5/commits/61a898bcb5004337c72133e74a74f95e439544cc